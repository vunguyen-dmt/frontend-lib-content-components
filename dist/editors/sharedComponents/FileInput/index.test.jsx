import React from 'react';
import { mount } from 'enzyme';

import { FileInput } from '.';

describe('FileInput component', () => {
  let el;
  let container;
  let props;
  beforeEach(() => {
    container = {};
    props = {
      acceptedFiles: '.srt',
      fileInput: {
        addFile: jest.fn().mockName('props.fileInput.addFile'),
        ref: (input) => { container.ref = input; },
      },
    };
    el = mount(<FileInput {...props} />);
  });
  test('snapshot', () => {
    expect(el).toMatchSnapshot();
  });
  test('only accepts allowed file types', () => {
    expect(el.find('input').props().accept).toEqual('.srt');
  });
  test('calls fileInput.addFile onChange', () => {
    expect(el.find('input').props().onChange).toEqual(props.fileInput.addFile);
  });
  test('loads ref from fileInput.ref', () => {
    expect(container.ref).toEqual(el.find('input').getDOMNode());
  });
});
