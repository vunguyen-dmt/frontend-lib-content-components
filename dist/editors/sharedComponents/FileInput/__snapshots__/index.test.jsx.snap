// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileInput component snapshot 1`] = `
<FileInput
  acceptedFiles=".srt"
  fileInput={
    Object {
      "addFile": [MockFunction props.fileInput.addFile],
      "ref": [Function],
    }
  }
>
  <input
    accept=".srt"
    className="upload d-none"
    onChange={[MockFunction props.fileInput.addFile]}
    type="file"
  />
</FileInput>
`;
